apps/src/templates/teacherNavigation/TeacherNavigationBar.tsx

```diff
@@ -56,78 +71,69 @@ const TeacherNavigationBar: React.FunctionComponent = () => {
-      isSelected={false}
-    />,
-    <SidebarOption
-      icon={'calendar'}
-      optionTitle={i18n.calendar()}
-      isSelected={false}
-    />,
+  const coursecontentSectionTitle = getSectionHeader(i18n.courseContent());
+  const courseContentKeys: (keyof typeof LABELED_TEACHER_NAVIGATION_PATHS)[] = [
+    'courseOverview',
+    'lessonMaterials',
+    'calendar',
   ];
 
-  const performanceContent = [
-    getSectionHeader(i18n.performance()),
-    <SidebarOption
-      icon={'chart-line'}
-      optionTitle={LABELED_TEACHER_NAVIGATION_PATHS.progress.label}
-      isSelected={false}
-    />,
-    <SidebarOption
-      icon={'star'}
-      optionTitle={LABELED_TEACHER_NAVIGATION_PATHS.assessments.label}
-      isSelected={false}
-    />,
-    <SidebarOption
-      icon={'code'}
-      optionTitle={LABELED_TEACHER_NAVIGATION_PATHS.projects.label}
-      isSelected={true}
-    />,
-    <SidebarOption
-      icon={'chart-simple'}
-      optionTitle={LABELED_TEACHER_NAVIGATION_PATHS.stats.label}
-      isSelected={false}
-    />,
-    <SidebarOption
-      icon={'pen-line'}
-      optionTitle={LABELED_TEACHER_NAVIGATION_PATHS.textResponses.label}
-      isSelected={false}
-    />,
-  ];
+  const performanceSectionTitle = getSectionHeader(i18n.performance());
+  const performanceContentKeys: (keyof typeof LABELED_TEACHER_NAVIGATION_PATHS)[] =
+    ['progress', 'assessments', 'projects', 'stats', 'textResponses'];
+
+  const classroomContentSectionTitle = getSectionHeader(i18n.classroom());
+  const classroomContentKeys: (keyof typeof LABELED_TEACHER_NAVIGATION_PATHS)[] =
+    ['manageStudents', 'settings'];
 
-  const classroomContent = [
-    getSectionHeader(i18n.classroom()),
-    <SidebarOption
-      icon={'users'}
-      optionTitle={LABELED_TEACHER_NAVIGATION_PATHS.manageStudents.label}
-      isSelected={false}
-    />,
-    <SidebarOption
-      icon={'gear'}
-      optionTitle={i18n.settings()}
-      isSelected={false}
-    />,
+  const teacherNavigationBarContent = [
+    {title: coursecontentSectionTitle, keys: courseContentKeys},
+    {title: performanceSectionTitle, keys: performanceContentKeys},
+    {title: classroomContentSectionTitle, keys: classroomContentKeys},
   ];
 
+  const getSidebarOptionsForSection = (sidebarKeys: string[]) => {
+    return sidebarKeys.map(key => (
+      <SidebarOption
+        key={'ui-test-sidebar-' + key}
+        isSelected={selectedOptionKey === key}
+        sectionId={+selectedSectionId}
+        pathKey={key as keyof typeof LABELED_TEACHER_NAVIGATION_PATHS}
+        onClick={() => handleOptionClick(key)}
+      />
+    ));
+  };
+
+  const navbarComponents = teacherNavigationBarContent.map(
+    ({title, keys}, index) => {
+      const sidebarOptions = getSidebarOptionsForSection(keys);
+
+      return (
+        <div key={`section-${index}`}>
+          {title}
+          {sidebarOptions}
+        </div>
+      );
+    }
+  );
+
+  const handleDropdownChange = (
```

Comment:
This is how I was doing it: https://github.com/code-dot-org/code-dot-org/blob/lfm/n-loader/apps/src/templates/teacherNavigation/TeacherNavigationBar.tsx#L158

Reply:
Feel free to ignore all the section switching dropdown stuff. I was not clear, but that is what I mean by the section loader I am working on

Reply:
I will leave for now, but thanks for the lead!